net/connUDP.go

```diff
@@ -16,13 +17,11 @@ import (
 //
 // Multiple goroutines may invoke methods on a UDPConn simultaneously.
 type UDPConn struct {
-	heartBeat      time.Duration
-	connection     *net.UDPConn
-	packetConn     packetConn
-	errors         func(err error)
-	network        string
-	onReadTimeout  func() error
-	onWriteTimeout func() error
+	connection *net.UDPConn
+	packetConn packetConn
+	errors     func(err error)
+	network    string
+	closed     uint32
```

Comment:
Using `atomic.Bool` would be safer, somebody could use non-atomic operations with `uint32` by mistake. Non-atomic comparisons are disabled for `atomic` variables.